### DISCUSSION TITLE
Fetch data across collections based on filters

### DISCUSSION THREAD

**Discussion by impondesk** [withastro/roadmap#976] [Proposal]
### Body

# Summary

We may need to fetch data across colllections, may be top 1 or top 2 entries across all collection types or selected collection type.

# Background & Motivation

For landing page, we may need aggregated data based on different collections and datasets.
We may fetch any combination of data data based on algorithm / filters to fetch a dynamic landing page.

# Goals

- Dynamic Newsletter page

# Example

E.g. https://blog.google/ and https://www.elegantthemes.com/preview/Extra/

**Comment by florian-lefebvre** [comment]
What is your request exactly? You can already do aggregation yourself by making several `getCollection`/`getEntry` calls

**Reply by hbeneke** [reply]
Hi @florian-lefebvre, I'm having a similar problem with content fetching. 

I have a list of notes, but I only need to get the latest note in my specific case. Assuming the list of notes is already ordered by date, how can I get only the last/first one?

Right now I'm doing ```const allNotes: Note[] = notes.flatMap((item) => item.data.notes);``` and then I'm looking for the first/last one on the list.
Is there any way to get only 1 element with a specific filter similar to what's discussed in this thread?

Thanks

**Reply by florian-lefebvre** [reply]
Not as of now, you need to get everything and filter in JS. What you ask for would probably require the content layer to be backed by a database (cc @ascorbic)
